src/vorta/application.py

```diff
@@ -64,6 +56,17 @@ def __init__(self, args_raw, single_app=False):
         self.backup_cancelled_event.connect(self.backup_cancelled_event_response)
         self.message_received_event.connect(self.message_received_event_response)
         self.set_borg_details_action()
+        self.installEventFilter(self)
+
+    def eventFilter(self, source, event):
+        if event.type() == QtCore.QEvent.ApplicationPaletteChange and type(source) == MainWindow:
+            self.main_window.set_icons()
+            self.main_window.repoTab.set_icons()
+            self.main_window.archiveTab.set_icons()
+            self.main_window.scheduleTab.set_icons()
+        if event.type() == QtCore.QEvent.ApplicationPaletteChange and source == self.tray.contextMenu():
+            self.tray.set_tray_icon()
```

Comment:
Is this switch over the source widget necessary? I suppose that wouldn't hurt to update the icons on every palette change.